### ISSUE TITLE
Issue with nested conditional ternary operator

### ISSUE THREAD

**Issue by priyatham-aps** [jscs-dev/babel-jscs#15]
I have the following code snippet:

``` javascript
let toogleStatus = node.removed ?
    (self.state.addNode !== '' ? !(self.state.addNode.split('_')[0] === node.id) : true)
    : (self.state.removeNode !== '' ? (self.state.removeNode.split('_')[0] === node.id) : false);
```

Running jscs throws the error "disallowMultipleSpaces: at most 1 spaces required between removeNode and : at filename.js". 

The rule I have configured for this is "disallowMultipleSpaces": true.

Running jscs with --fix option results in a very weird code:

``` javascript
let toogleStatus = node.removed ?
    (self.state.addNode !== '' ? !(self.state.addNode.split('_')[0] === node.id) ::: true)
    : (self.state.removeNode : (self.state.removeNode !== '' ? (self.state.removeNode.split('_')[0] === node.id) ::: false);
```

I am using master branch of node-jscs and babel-jscs version 2.0.0. Did not encounter the issue with other parsers like esprima or esprima-fb.

Any thoughts?

Thanks.

**Comment by hzoo** [comment]
Hmm I tried reproducing with your code snippet but don't get any errors? This is with babel-jscs 2.0.0, jscs#master and latest babel version as well (maybe update that).

**Comment by priyatham-aps** [comment]
I am using latest babel: 5.6.23 (babel-core 5.7.4)

I have created a minimal repo for the issue at https://github.com/priyatham-aps/babel-jscs-ternaryop-issue.

Please check it out, run npm install and jscs. Hopefully, this will help in demonstrating the issue.

**Comment by priyatham-aps** [comment]
I added a few more scenarios in the above repo. Looks like the issue is not with the nesting of ternary operators but usage of brackets in the expressions of ternary operator.

**Comment by priyatham-aps** [comment]
@hzoo 
Are you able to reproduce the issue? Or do you need any further help from me on this?

**Comment by hzoo** [comment]
Sorry been kinda busy so couldn't get to this. I just tried it again and cannot reproduce - what node version are you using? Also what you can do now is just do

``` js
  "devDependencies": {
    "jscs": "jscs-dev/node-jscs"
  }
```

because jscs will include `babel and babel-jscs` already.

In your `.jscsrc` you can do

``` js
{
    "esnext": true,
    "disallowMultipleSpaces": true
}
```

since it will exclude `node_modules` by default now and esnext will use `babel-jscs`

Or you can do `./node_modules/.bin/jscs index.js --esnext`. I didn't get any errors after doing this either (with io.js 2.3.4)

I just updated the readme with this - hopefully it's more useful now.

**Comment by priyatham-aps** [comment]
I just tried it the way you have recommended by uninstalling `babel-jscs` and `babel` from `dev-dependencies` and removing `esprima` option from `.jscsrc`. When I ran `jscs`, I got an error saying `Cannot find module 'babel-jscs'`. 

I then did `npm uninstall jscs --save-dev` and installed it again. Now, I am not getting the issue anymore. At this point, it looks like I may have been using an earlier commit from master branch of jscs, though I am not entirely sure.

Appreciate the info and help though. Thanks.
